js/components/navigationBar.js

```diff
@@ -51,8 +51,12 @@ class NavigationBar extends ImmutableComponent {
     if (!isBookmarked) {
       appActions.addSite(siteDetail, siteTags.BOOKMARK)
     }
+    // Show bookmarks toolbar after first bookmark is saved
     appActions.changeSetting(settings.SHOW_BOOKMARKS_TOOLBAR, !hasBookmark || showBookmarksToolbar)
+    // trigger the AddEditBookmark modal
     windowActions.setBookmarkDetail(siteDetail, siteDetail)
+    // Update checked/unchecked status in the Bookmarks menu
+    ipc.send(messages.UPDATE_MENU_BOOKMARKED_STATUS, this.bookmarked)
```

Comment:
similar to the issue with DATA_FOR_WINDOW, we're using ipc messages in a lot of places where we could (and should I think) just be using actions. The windowActions are already sent to the appStore via the dispatcher, we're just not handling them there. Ultimately this seems a little backwards to me anyway because bookmarks are part of appState, not windowState and the bookmarked state of the current url should come from there. I realize there are legacy reasons for all of this, but it would really be nice to clean some of it up.